pkg/helper/openapi_utils.go

```diff
@@ -0,0 +1,173 @@
+package helper
+
+import (
+	"bytes"
+	"fmt"
+	"net/url"
+	"regexp"
+	"strings"
+	"text/template"
+
+	"github.com/getkin/kin-openapi/openapi3"
+)
+
+var (
+	// NonWordCharRegexp not word characters (== [^0-9A-Za-z_])
+	NonWordCharRegexp = regexp.MustCompile(`\W`)
+
+	// TemplateRegexp used to render openapi server URLs
+	TemplateRegexp = regexp.MustCompile(`{([\w]+)}`)
+
+	// NonAlphanumRegexp not alphanumeric
+	NonAlphanumRegexp = regexp.MustCompile(`[^0-9A-Za-z]`)
+)
+
+func SystemNameFromOpenAPITitle(obj *openapi3.Swagger) string {
+	openapiTitle := obj.Info.Title
+	openapiTitleToLower := strings.ToLower(openapiTitle)
+	return NonWordCharRegexp.ReplaceAllString(openapiTitleToLower, "_")
+}
+
+func K8sNameFromOpenAPITitle(obj *openapi3.Swagger) string {
+	openapiTitle := obj.Info.Title
+	openapiTitleToLower := strings.ToLower(openapiTitle)
+	return NonAlphanumRegexp.ReplaceAllString(openapiTitleToLower, "")
```

Comment:
I'm probably nitpicking here but if K8s names have character length limitation. Maybe we should document somewhere that the title in the openapi spec has to have a max length.
K8s max length seems to be 253 characters which is quite high (https://kubernetes.io/docs/concepts/overview/working-with-objects/names/) but the names often end up automatically being part of a label for some objects, in which case they effectively have a maximum length of 63 characters which is easily reachable.
I found myself in that situation when working on backup/restore. I don't how likely is that to happen here though so letting it up to you.

Reply:
That comment already exists in functions calling this function. Those functions complete the final k8s name and therefore the comment was done there.

Reply:
Sorry, I meant on the user-facing side.
How likely is that it will reach this limit and thus an error creating objects will happen?

Reply:
The code verifies if it is valid DNS1123 Label. However, this is a k8s name. The validation is wrong. 

Can you elaborate more on your statement `but the names often end up automatically being part of a label for some objects`.

Reply:
In some K8s objects a label is created with the same value as the resource name which effectively means that the K8s name cannot be longer than the label limit. I think the case where I detected that  was in `Job` objects. That's why the validation was done more strict in that case.

Reply:
But that does not apply to products and backends created by openapi CR

I will change validation

Reply:
:+1: . Whatever the implementation is, if you think it will be likely for the limit to be reached we should add user-facing documentation. If it's not likely to happen then we are ok a doc level

Reply:
added user-facing doc about max length size of the OpenAPI title.

Reply:
:+1: